torchmetrics/text/mer.py

```diff
@@ -0,0 +1,109 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, List, Optional, Union
+from warnings import warn
+
+import torch
+from torch import Tensor, tensor
+
+from torchmetrics.functional.text.mer import _mer_compute, _mer_update
+from torchmetrics.metric import Metric
+
+
+class MatchErrorRate(Metric):
+    r"""
+    Match error rate (MatchErrorRate_) is a common metric of the performance of an automatic speech recognition system.
+    This value indicates the percentage of words that were incorrectly predicted and inserted.
+    The lower the value, the better the performance of the ASR system with a MatchErrorRate of 0 being a perfect score.
+    Match error rate can then be computed as:
+
+    .. math::
+        mer = \frac{S + D + I}{N + I} = \frac{S + D + I}{S + D + C + I}
+
+    where:
+        - S is the number of substitutions,
+        - D is the number of deletions,
+        - I is the number of insertions,
+        - C is the number of correct words,
+        - N is the number of words in the reference (N=S+D+C).
+
+    Compute MatchErrorRate score of transcribed segments against references.
+
+    Args:
+        concatenate_texts: Whether to concatenate all input texts or compute MatchErrorRate iteratively.
+            This argument is deprecated in v0.6 and it will be removed in v0.7.
```

Comment:
```suggestion
        concatenate_texts: Whether to concatenate all input texts or compute MatchErrorRate iteratively.
            This argument is deprecated in v0.6 and it will be removed in v0.7.
```
we are already in 0.6 so don't add when it is immediately deprecated

Reply:
ok